### ISSUE TITLE
unfubar strcat() in user_main.c

### ISSUE THREAD

**Issue by geekmaster** [cnlohr/esp82xx#52]
esp82xx/fwsrc/user/user_main.c strcat() calls itself?

"that's really wrong.  File a bug report.
that's like /really/ wrong.
It's because the standard library is missing strcat, I believe."
-- cnlohr

**Comment by con-f-use** [comment]
Yeah, it seems some magic code, I didn't write has injected itself into the last merge. No idea what happened there. Also the rf_cal functions and so on changed.

**Comment by cnlohr** [comment]
I vaguely recall putting in the strcat code.  Though I feel completely idiotic about writing it now.  I don't know what I was thinking.

charrx I think is when data comes back down the serial line.  That way you can easily hook incoming serial bytes.

**Comment by cnlohr** [comment]
P.S. Please use your code if you've tested it.  I also noticed 'man cat' has an implementation:

```
           char *
           strncat(char *dest, const char *src, size_t n)
           {
               size_t dest_len = strlen(dest);
               size_t i;

               for (i = 0 ; i < n && src[i] != '\0' ; i++)
                   dest[dest_len + i] = src[i];
               dest[dest_len + i] = '\0';

               return dest;
           }
```

**Comment by con-f-use** [comment]
Fixed in 93074f1

**Comment by cnlohr** [comment]
and it works :-D
